spec/controllers/api/drafts_controller_spec.rb

```diff
@@ -79,6 +79,60 @@
     assert_equal(parsed_body['error'], 'unauthorized')
   end
 
+  it 'can not update a draft if the user does not belong to the provider list.' do
+    # The draft is created by a MMT_2 user
+    # The testeruser2 updating the document does not have MMT_2 in their available provider list, only 'LARC'
+    allow_any_instance_of(Cmr::UrsClient).to receive(:validate_mmt_token).and_return(Faraday::Response.new(status: 200, body: '{"uid":"testuser2"}', response_headers: { 'Content-Type': 'application/json; charset=utf-8' }))
+    jsonContent = { "json": {
+      "Name": "a name",
+      "LongName": "a long name",
+      "Version": "10.0"
+    }}.to_json
+    put :update, body: jsonContent, params: { id: @tool_draft.id, draft_type: "ToolDraft" }
+    assert_equal(response.status, 401)
+    parsed_body = JSON.parse(response.body)
+    assert_equal(parsed_body['error'], 'unauthorized')
+  end
+
+  it 'create a draft record.' do
```

Comment:
So you have a test to prevent updating a draft record with a illegal provider, but not create.   You'll need to do something like this test but with a illegal provider instead.